core/domain/user_services_test.py

```diff
@@ -2217,6 +2218,420 @@ def test_flush_migration_bot_contributions_model(self):
+        self.assertDictEqual(
+            stats.translation_reviewer_counts_by_lang_code,
+            {'hi': 1, 'en': 1})
+
+    def test_grant_reviewer_existing_translation_reviewing_rights_no_count_diff(
+            self):
+        self._assert_community_contribution_stats_is_in_default_state()
+        user_services.allow_user_to_review_translation_in_language(
+            self.reviewer_1_id, 'hi')
+        # Assert that the translation reviewer count increased by one.
+        stats = suggestion_services.get_community_contribution_stats()
+        self.assertEqual(stats.question_reviewer_count, 0)
+        self.assertEqual(stats.question_suggestion_count, 0)
+        self.assertDictEqual(
+            stats.translation_suggestion_counts_by_lang_code, {})
+        self.assertDictEqual(
+            stats.translation_reviewer_counts_by_lang_code, {'hi': 1})
+
+        user_services.allow_user_to_review_translation_in_language(
+            self.reviewer_1_id, 'hi')
+
+        # Assert that the translation reviewer count did not change because the
+        # reviewer already had the permissions.
+        stats = suggestion_services.get_community_contribution_stats()
+        self.assertEqual(stats.question_reviewer_count, 0)
+        self.assertEqual(stats.question_suggestion_count, 0)
+        self.assertDictEqual(
+            stats.translation_suggestion_counts_by_lang_code, {})
+        self.assertDictEqual(
+            stats.translation_reviewer_counts_by_lang_code, {'hi': 1})
+
+    def test_remove_all_reviewer_translation_reviewing_rights_decreases_count(
+            self):
+        self._assert_community_contribution_stats_is_in_default_state()
+        user_services.allow_user_to_review_translation_in_language(
+            self.reviewer_1_id, 'hi')
+        # Assert that the translation reviewer count increased by one.
+        stats = suggestion_services.get_community_contribution_stats()
+        self.assertEqual(stats.question_reviewer_count, 0)
+        self.assertEqual(stats.question_suggestion_count, 0)
+        self.assertDictEqual(
+            stats.translation_suggestion_counts_by_lang_code, {})
+        self.assertDictEqual(
+            stats.translation_reviewer_counts_by_lang_code, {'hi': 1})
+
+        user_services.remove_translation_review_rights_in_language(
+            self.reviewer_1_id, 'hi')
+
+        # Assert that the translation reviewer count decreased by one after the
+        # rights were removed.
+        stats = suggestion_services.get_community_contribution_stats()
+        self.assertEqual(stats.question_reviewer_count, 0)
+        self.assertEqual(stats.question_suggestion_count, 0)
+        self.assertDictEqual(
+            stats.translation_suggestion_counts_by_lang_code, {})
+        self.assertDictEqual(
+            stats.translation_reviewer_counts_by_lang_code, {'hi': 0})
+
+    def test_remove_reviewer_translation_reviewing_rights_decreases_count(
```

Comment:
Maybe: ``..._remove_some_reviewer_translation_reviewing_rights...`` to contrast a bit more clearly with the previous test.

Reply:
Good idea, thanks :)